pkg/logcli/query/query.go

```diff
@@ -157,9 +195,195 @@ func (q *Query) DoQuery(c client.Client, out output.LogOutput, statistics bool)
 				// fudge the timestamp forward in time to make sure to get the last entry from this batch in the next query
 				end = lastEntry[0].Timestamp.Add(1 * time.Nanosecond)
 			}
+		}
+	}
+
+	if partFile != nil {
+		if err := partFile.Complete(); err != nil {
+			log.Fatalln(err)
+		}
+	}
+}
+
+func (q *Query) outputFilename() string {
+	return fmt.Sprintf("%s-%d-%d.part", q.PartPrefix, q.Start.Unix(), q.End.Unix())
+}
+
+// createPartFile returns a PartFile if the PartFilePrefix is set.
+// The bool value shows if the part file already exists, and this range should be skipped.
+func (q *Query) createPartFile() (*PartFile, bool) {
```

Comment:
It was a little confusing to me when I read that we always use a part file. What do you think about `createOptionalPartFile`?

Reply:
I have moved the conditional creation to the caller, so this is much clearer now.